test/sys/test_socket.rs

```diff
@@ -186,6 +186,57 @@ pub fn test_scm_rights() {
     close(w).unwrap();
 }
 
+#[cfg(any(target_os = "linux", target_os= "android"))]
+#[test]
+pub fn test_af_alg_cmsg() {
+    use std::{mem, ptr};
+    use nix::sys::uio::IoVec;
+    use nix::unistd::read;
+    use nix::sys::socket::{socket, sendmsg, bind, accept, setsockopt,
+                           AddressFamily, SockType, SockFlag, SockAddr,
+                           ControlMessage, MsgFlags};
+    use nix::sys::socket::sockopt::AlgSetKey;
+
+    let alg_type = "skcipher";
+    let alg_name = "ctr(aes)";
+
+    let sock = socket(AddressFamily::Alg, SockType::SeqPacket, SockFlag::empty(), None)
+        .expect("socket failed");
+
+    let sockaddr = SockAddr::new_alg(alg_type, alg_name);
+    bind(sock, &sockaddr).expect("bind failed");
+    let key = vec![1u8, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16,
+                   17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32];
+
+    setsockopt(sock, AlgSetKey, &key).expect("setsockopt");
+    let session_socket = accept(sock).expect("accept failed");
+
+    let iv = vec![1u8, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
+    let mut iv_storage = vec![0u8; 16 + mem::size_of::<libc::af_alg_iv>()];
+    let alg_iv: *mut libc::af_alg_iv = unsafe { mem::transmute(iv_storage.as_mut_ptr()) };
```

Comment:
`unsafe` sets of alarm bells.  `mem::transmute` sets of loud alarm bells.  We shouldn't be creating an API that requires the use of `mem::transmute`.  Can you eliminate it?  Ideally Nix would only expose safe APIs.

Reply:
Ok, makes sense. So, AlgSetIv should accept bytes, and perform unsafe logic inside